src/UI/WB.UI.Designer/Controllers/Api/Designer/AttachmentsController.cs

```diff
@@ -11,53 +11,82 @@
 using SixLabors.ImageSharp.Processing;
 using WB.Core.BoundedContexts.Designer.Implementation.Services.AttachmentService;
 using WB.Core.BoundedContexts.Designer.Services;
+using WB.Core.BoundedContexts.Designer.Views.Questionnaire.ChangeHistory;
 using WB.UI.Designer.Extensions;
 
 namespace WB.UI.Designer.Controllers.Api.Designer
 {
     [Route("attachments")]
-    [Authorize]
+    [AuthorizeOrAnonymousQuestionnaire]
+    [QuestionnairePermissions]
     public class AttachmentsController : Controller
     {
         private const int defaultImageSizeToScale = 156;
 
         private readonly IAttachmentService attachmentService;
         private readonly IWebHostEnvironment environment;
+        private readonly IDesignerQuestionnaireStorage questionnaireStorage;
 
         public AttachmentsController(IAttachmentService attachmentService,
-            IWebHostEnvironment environment)
+            IWebHostEnvironment environment,
+            IDesignerQuestionnaireStorage questionnaireStorage)
         {
             this.attachmentService = attachmentService;
             this.environment = environment;
+            this.questionnaireStorage = questionnaireStorage;
         }
 
         [HttpGet]
-        [Route("{id:Guid}")]
-        public IActionResult Get(Guid id)
+        [Route("{id}/{attachmentId:Guid}")]

```

Comment:
Are we keeping backward compatibility with previously generated htmls?

Reply:
Yes, I added old methods